include/CL/opencl.hpp

```diff
@@ -10400,6 +10421,27 @@ namespace compatibility {
 } // namespace compatibility
 
 #ifdef cl_khr_semaphore
+
+#ifdef cl_khr_external_semaphore
+enum class ExtSemaphoreHandleTypeKhr : cl_external_semaphore_handle_type_khr
+{
+    TypeNone = 0,
+#ifdef cl_khr_external_semaphore_dx_fence
+    TypeD3D12Fence = CL_SEMAPHORE_HANDLE_D3D12_FENCE_KHR,
+#endif
```

Comment:
There is a little bit of naming redundancy here and some inconsistency with the recently added `ExternalMemoryType` enum.

How do you feel about:

```suggestion
enum class ExternalSemaphoreType : cl_external_semaphore_handle_type_khr
{
    None = 0,
#ifdef cl_khr_external_semaphore_dx_fence
    D3D12Fence = CL_SEMAPHORE_HANDLE_D3D12_FENCE_KHR,
#endif
```

etc.

Key changes are:
* Spell out `External`.
* Drop `Handle` and `Khr` from the enum type name.
* Drop `Type` from the enums themselves.

This means we will write `ExternalSemaphoreType::D3D12Fence` rather than `ExtSemaphoreHandleTypeKhr::TypeD3D12Fence`, which seems shorter and more descriptive.

Reply:
Done, thanks.